torch/distributed/tensor/parallel/api.py

```diff
@@ -67,8 +68,15 @@ def parallelize_module(  # type: ignore[return]
     """
     torch._C._log_api_usage_once("torch.distributed.tensor.parallel.parallelize_module")
 
+    device_mesh = device_mesh or _mesh_resources.get_current_mesh()
```

Comment:
If we have nD DeviceMesh, what this line of code will retrieve? Would it intelligently fetch the TP mesh?

Reply:
Whether the code is:
```
parallelize_module(mod, mesh, plan)
```
or 
```
with mesh:
    parallelize_module(mod, plan)
```
the user has responsibility to make sure `mesh` is a correct, intended one.

Reply:
If `mesh` is wrong, the above two pieces of code would also fail in the same way.

Reply:
this makes sense to me.

but would this code run? I thought it at least should be:
`parallelize_module(mod, parallelize_plan=plan)` which doesn't make it that simple compared with `parallelize_module(mod, mesh, plan)`